### DISCUSSION TITLE
Aula EF CORE relacionamento muitos para muitos

### DISCUSSION THREAD

**Discussion by CarlosDoScript** [balta-io/forum#210] [Tira Dúvidas] [answered]
gostaria de sabe se nesse codigo 
                 builder
                .HasMany(x => x.Roles)
                .WithMany(x => x.Users)
                .UsingEntity<Dictionary<string, object>>(
                    "UserRole",
                    role => role
                        .HasOne<Role>()
                        .WithMany()
                        .HasForeignKey("RoleId")
                        .HasConstraintName("FK_UserRole_RoleId")
                        .OnDelete(DeleteBehavior.Cascade),
                    user => user
                        .HasOne<User>()
                        .WithMany()
                        .HasForeignKey("UserId")
                        .HasConstraintName("FK_UserRole_UserId")
                        .OnDelete(DeleteBehavior.Cascade));

tem como adicionar um campo do tipo bit com o nome deletado para caso eu venha deletar so deixar o campo como true

**Comment by andrebaltieri** [comment, marked as answer]
Boa tarde, @CarlosDoScript, como vai?

Neste código em específico não, pois o mapeamento está sendo feito de forma oculta. Neste caso a tabela associativa vai ser gerada automaticamente.

Agora você pode criar uma entidade UserRole com o campo que quer ter e fazer os mapeamentos nela.
